packages/siopv2-oid4vp-rp-rest-client/src/agent/SiopV2OID4VpRpRestClient.ts

```diff
@@ -0,0 +1,99 @@
+import { fetch } from 'cross-fetch'
+import {
+  ISiopV2OID4VpRpRestClient,
+  ISiopClientGenerateAuthRequestArgs,
+  ISiopClientGetAuthStatusArgs,
+  IRequiredContext,
+  ISiopClientRemoveAuthRequestSessionArgs,
+} from '../index'
+import Debug from 'debug'
+import { IAgentPlugin } from '@veramo/core'
+import { AuthStatusResponse, GenerateAuthRequestURIResponse } from '@sphereon/ssi-sdk-siopv2-oid4vp-common'
+
+const debug = Debug('ssi-sdk-siopv2-oid4vp-rp-rest-client:SiopV2OID4VpRpRestClient')
+
+export class SiopV2OID4VpRpRestClient implements IAgentPlugin {
+  readonly methods: ISiopV2OID4VpRpRestClient = {
+    siopClientRemoveAuthRequestSession: this.siopClientRemoveAuthRequestSession.bind(this),
+    siopClientGenerateAuthRequest: this.siopClientGenerateAuthRequest.bind(this),
+    siopClientGetAuthStatus: this.siopClientGetAuthStatus.bind(this),
+  }
+
+  private readonly baseUrl?: string
+  private readonly definitionId?: string
+
+  constructor(baseUrl?: string, definitionId?: string) {
+    if (baseUrl) {
+      this.baseUrl = baseUrl
+    }
+    if (definitionId) {
+      this.definitionId = definitionId
+    }
+  }
+
+  private async siopClientRemoveAuthRequestSession(args: ISiopClientRemoveAuthRequestSessionArgs, context: IRequiredContext): Promise<void> {
+    const baseUrl = this.checkBaseUrlParameter(args.baseUrl)
+    const definitionId = this.checkDefinitionIdParameter(args.definitionId)
+    fetch(this.uriWithBase(`/webapp/definitions/${definitionId}/auth-requests/${args.correlationId}`, baseUrl), {
```

Comment:
Isn't this returning a response? This is a client, so it should return the API responses

Reply:
in the current code no
```
this.express.delete("/webapp/definitions/:definitionId/auth-requests/:correlationId", (request, response) => {
                const correlationId: string = request.params.correlationId
                this.verifier.sessionManager.deleteStateForCorrelationId(correlationId)
            }
        )
```